[docs] update db guide introductions

GavinFrazar opened the pull request:
This PR updates the introduction snippets for database guides.

I simplified the snippets and removed some redundant parameters.

I also changed the cockroach and redis cluster guides to use the "self hosted" variant of the intro.

GavinFrazar committed:
update db guide introductions

GavinFrazar commented:
my that's a lot of vale lint errors!

A lot of them seem to be already in the file though, e.g. the error about the vitess guide using "Teleport Enterprise Cloud" here:

```
<Tabs>
<TabItem scope={["oss", "enterprise"]} label="Self-Hosted">
![Teleport Database Access Vitess Self-Hosted](../../../img/database-access/guides/vitess_selfhosted.png)
</TabItem>
<TabItem scope={["cloud"]} label="Teleport Enterprise Cloud">
![Teleport Database Access Vitess Cloud](../../../img/database-access/guides/vitess_cloud.png)
</TabItem>
```

I'll take a look at resolving them next week. @ptgott it says these are errors but I wonder if this check is not required or if they're not really "error" level?

greedy52 commented on docs/pages/database-access/guides/aws-cassandra-keyspaces.mdx:
```diff
@@ -3,11 +3,11 @@ title: Database Access with Amazon Keyspaces (Apache Cassandra)
 ---
 
-(!docs/pages/includes/database-access/db-introduction.mdx  dbType="Amazon Keyspaces (Apache Cassandra)" dbConfigure="Amazon Keyspaces database with IAM authentication" dbName="Amazon Keyspaces" !)
+(!docs/pages/includes/database-access/db-introduction.mdx dbType="AWS Keyspaces (Apache Cassandra)" dbConfigure="with IAM authentication"!)
```
Several lint including this one is correct on product name. It should be Amazon Keyspaces, Amazon ElastiCache, etc.

greedy52 commented on docs/pages/database-access/guides/aws-cassandra-keyspaces.mdx:
```diff
@@ -3,11 +3,11 @@ title: Database Access with Amazon Keyspaces (Apache Cassandra)
 
 ## How it works
 
-(!docs/pages/includes/database-access/how-it-works/iam.mdx db="Amazon Keyspaces" cloud="AWS"!)
```
why not keep the `Amazon` part?

GavinFrazar replied:
I thought it was redundant because of cloud=AWS, but I see that the snippet is fine with both. I'll revert it to keep the `Amazon` bit (and the others that I changed `Amazon -> AWS`)

greedy52 commented on docs/pages/database-access/guides/azure-postgres-mysql.mdx:
```diff
@@ -3,7 +3,7 @@ title: Database Access with Azure PostgreSQL and MySQL
 ---
 
-(!docs/pages/includes/database-access/db-introduction.mdx  dbType="Azure PostgreSQL or MySQL" dbConfigure="Azure PostgreSQL or MySQL database with IAM authentication" dbName="Azure PostgreSQL or MySQL" !)
+(!docs/pages/includes/database-access/db-introduction.mdx dbType="Azure PostgreSQL or MySQL" dbConfigure="with IAM authentication"!)
```
`dbConfigure` should be `with Microsoft Entra ID-based authentication` or `with Microsoft Entra ID for authentication`
https://learn.microsoft.com/en-us/azure/postgresql/single-server/how-to-configure-sign-in-azure-ad-authentication

Same applies to Azure Cache for Redis:
https://learn.microsoft.com/en-us/azure/azure-cache-for-redis/cache-azure-active-directory-for-authentication

I believe we already updated the "how it works" section below on this though.

greedy52 commented on docs/pages/database-access/guides/azure-sql-server-ad.mdx:
```diff
@@ -3,7 +3,7 @@ title: Database Access with SQL Server on Azure
 ---
 
-(!docs/pages/includes/database-access/db-introduction.mdx  dbType="Azure SQL Server" dbConfigure="Azure SQL Server using Azure Active Directory authentication" dbName="Azure SQL Server" !)
+(!docs/pages/includes/database-access/db-introduction.mdx dbType="Azure SQL Server" dbConfigure="using Azure Active Directory authentication"!)
```
nit: Azure Active Directory is renamed to Microsoft Entra ID
https://learn.microsoft.com/en-us/azure/azure-sql/database/authentication-aad-overview?view=azuresql

greedy52 reviewed (commented):
🎉 thanks !

ptgott commented:
> I'll take a look at resolving them next week. ptgott it says these are errors but I wonder if this check is not required or if they're not really "error" level?

@GavinFrazar Technically, the vale job doesn't need to run successfully for a docs PR to be merged. That might change one day, but I'd need to review leveling and potential false positives first. I'd pay attention to any errors related to consistent terms, though, since we're trying to roll out changes like, for example, "cloud-hosted Teleport Enterprise" in place of "Teleport Enterprise Cloud" to imply that there is a single Enterprise Teleport offering.

GavinFrazar commented:
Updated: https://github.com/gravitational/teleport/compare/d51d969d1b19e66fb97e8d6a87818ca64eb91132..ceabc6f15c618e506a23b0abdcb41ef23a70c750

- reverted all the changes that converted `AWS -> Amazon` phrasing
- converted usage of "Azure AD" to "Microsoft Entra ID" per @greedy52 's suggestion 👍 

And I realized I can see all the lint errors introduced by my PR are annotated, so the other 100+ errors were already there.
I'm going to make a pass at fixing those errors in a follow-up PR(s).
I'd like to keep this PR and follow-ups really simple so we don't have to review giant docs PRs
